index.js

```diff
@@ -355,34 +351,40 @@ async function handleSensorRequest(currentLocationId){
 
             if(typeof latestSession !== 'undefined'){ // Checks if no session exists for this location yet.
                 if(latestSession.end_time == null){ // Checks if session is open.
-                    await db.updateSessionState(latestSession.sessionid, currentState, currentLocationId);
+                    await db.updateSessionState(latestSession.sessionid, currentState);
                 }
             }
         }
 
 
         // Checks if current state belongs to the session triggerStates
         else if(TRIGGERSTATES.includes(currentState)){
-            let latestSession = await db.getMostRecentSession(currentLocationId);
+
+            let client = await db.beginTransaction()
+            let latestSession = await db.getMostRecentSession(currentLocationId, client);
 
             if(typeof latestSession !== 'undefined'){ //Checks if session exists
                 if(latestSession.end_time == null){  // Checks if session is open for this location
-                    let currentSession = await db.updateSessionState(latestSession.sessionid, currentState, currentLocationId);
+                    let currentSession = await db.updateSessionState(latestSession.sessionid, currentState, client);
                     start_times[currentLocationId] = currentSession.start_time;
                 }
                 else{
-                    let currentSession = await db.createSession(location.phonenumber, currentLocationId, currentState); // Creates a new session
+                    let currentSession = await db.createSession(location.phonenumber, currentLocationId, currentState, client); // Creates a new session
                     start_times[currentLocationId] = currentSession.start_time;
                 }
+            }else{
+                let currentSession = await db.createSession(location.phonenumber, currentLocationId, currentState, client); // Creates a new session
+                start_times[currentLocationId] = currentSession.start_time;
             }
+            await db.commitTransaction(client)
         }
 
         // Checks if current state belongs to the session closingStates
         else if(CLOSINGSTATES.includes(currentState)){
             let latestSession = await db.getMostRecentSession(currentLocationId);
-            await db.updateSessionState(latestSession.sessionid, currentState, currentLocationId); //Adds the closing state to session
+            await db.updateSessionState(latestSession.sessionid, currentState); //Adds the closing state to session
```

Comment:
If trigger states require a transaction to prevent race conditions, would closing states also go by the same logic, or are things different here?

Reply:
we have added a transaction to closing states as well! I think you are looking at an old commit here